go/driver.go

```diff
@@ -0,0 +1,82 @@
+package openmldb
+
+import (
+	"context"
+	"database/sql"
+	interfaces "database/sql/driver"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+func init() {
+	sql.Register("openmldb", &driver{})
+}
+
+var (
+	_ interfaces.Driver        = (*driver)(nil)
+	_ interfaces.DriverContext = (*driver)(nil)
+
+	_ interfaces.Connector = (*connecter)(nil)
+)
+
+type driver struct{}
+
+func parseDsn(dsn string) (host string, db string, err error) {
+	u, err := url.Parse(dsn)
+	if err != nil {
+		return "", "", fmt.Errorf("invlaid URL: %w", err)
+	}
+
+	if u.Scheme != "openmldb" && u.Scheme != "" {
+		return "", "", fmt.Errorf("invalid URL: unknown schema '%s'", u.Scheme)
+	}
+
+	p := strings.Split(strings.TrimLeft(u.Path, "/"), "/")
+
+	if len(p) == 0 {
+		return "", "", fmt.Errorf("invalid URL: DB name not found")
```

Comment:
does api server support connection without db ?

Reply:
I think no, at least for now. because apiserver has no rest endpoint that does not need db